com.unity.multiplayer.mlapi/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -251,12 +251,16 @@ private void OnSceneUnloadServer(Guid switchSceneGuid)
             var newSceneObjects = new List<NetworkObject>();
             {
                 var networkObjects = UnityEngine.Object.FindObjectsOfType<NetworkObject>();
+
                 for (int i = 0; i < networkObjects.Length; i++)
                 {
-                    if (networkObjects[i].IsSceneObject == null)
+                    if (networkObjects[i].NetworkManager == m_NetworkManager)
```

Comment:
I was looking at this and wondering how this works in practice if you have 2 NM's.  I get how you override the NM for a NO.  How about for a scene?  Or are we saying that just one NM will ever have the scene-defined objects?  Then I wonder if the contrary of this is true, is it an error?  That is you have a NO that is a scene object that belongs to the NM that is not the one that handles scene objects.  If that's true (I know this is a long walk...) then seems like we should throw an exception here.